packages/headless/src/app/case-assist-engine/case-assist-engine.test.ts

```diff
@@ -86,6 +86,43 @@ describe('buildCaseAssistEngine', () => {
       };
       expect(initEngine).toThrow();
     });
+
+    describe('the analytics mode', () => {
+      let warnSpy: ReturnType<typeof vi.spyOn>;
+
+      beforeEach(() => {
+        warnSpy = vi.spyOn(console, 'warn').mockImplementation(() => {});
+      });
+
+      afterEach(() => {
+        warnSpy.mockRestore();
+      });
+
+      it('should log a warning when the case assist engine is used with the next analytics mode', () => {
+        options.configuration.analytics = {
+          analyticsMode: 'next',
+          trackingId: 'example_tracking_id',
+        };
+        initEngine();
+
+        expect(warnSpy).toHaveBeenCalledTimes(1);
+        expect(warnSpy).toHaveBeenCalledWith(
+          '[Warning] A component from the Coveo Headless library has been instantiated with the Analytics Mode: "Next".\n' +
```

Comment:
Can you think of a place where we could store that string so that both the engines import it to display it if needed, but also we could import it in the tests for the expects?

I'm just trying to see in the future when we'll want to perhaps modify or remove it, it might be simpler if the text is just in one place instead of copied everywhere.

I would say if we don't already have a place where we contain "texts" in Headless, then nevermind, I wouldn't create a new pattern just for this.